### PR TITLE
Add graph view for weather

### PR DESCRIPTION
**Pull request by AndrewHelix**
Screenshot 2022-09-22 182028
Screenshot 2022-09-22 182056

### CODE REVIEW

File: src/components/main/GraphDaily.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import React, { FC, useContext, useMemo } from 'react';
+import { Line } from 'react-chartjs-2';
+import { Chart, registerables } from 'chart.js';
+Chart.register(...registerables);
```
Comment (JuliaShavyrova) [inline src/components/main/GraphDaily.tsx]: please follow ordering - imports first, then code

File: src/components/main/GraphDaily.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+type GraphDailyProps = {
+  weather: DailyWeatherType[];
+};
+
+const GraphDaily: FC<GraphDailyProps> = ({ weather }) => {
+  const { options } = useGraphSettings();
+
+  const data = {
+    labels: weather.map(el => `${convertTime(el.date).date}-${convertTime(el.date).month}`),
```
Comment (JuliaShavyrova) [inline src/components/main/GraphDaily.tsx]: please move to separate function

File: src/components/main/GraphHourly.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+type GraphHourlyProps = {
+  weather: HourlyWeatherType[];
+};
+
+const GraphHourly: FC<GraphHourlyProps> = ({ weather }) => {
+  const { options } = useGraphSettings();
+
+  const data = {
+    labels: weather.map(el => `${convertTime(el.time).hours}-${convertTime(el.time).minutes}`),
```
Comment (JuliaShavyrova) [inline src/components/main/GraphHourly.tsx]: the same about function

File: src/components/main/Main.tsx [hunk 4]
```diff
@@ -65,13 +68,37 @@ const Main: FC = () => {
       {currentWeather ? (
         <>
           <MainCard currentWeather={currentWeather} location={location} />
+          <div className={styles['view-btns-wrapper']}>
+            <button
+              className={styles['view-btn']}
+              onClick={() => setView('cards')}
+              style={{
+                backgroundColor: view === 'cards' ? '#ffff00' : '#F8F8FF'
```
Comment (JuliaShavyrova) [inline src/components/main/Main.tsx]: please create additional selectors and move styles inside it

File: src/hooks/useGraphSettings.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import { useContext, useMemo } from 'react';
+import { themeContext } from 'context/themeContext';
+
+export const useGraphSettings = () => {
```
Comment (JuliaShavyrova) [inline src/hooks/useGraphSettings.ts]: please add type definitions

**Comment by AndrewHelix** [comment]
Change graph
In the second graph i don't change color for lines before and after zero for easy different between day and night
Screenshot 2022-09-23 163112
Screenshot 2022-09-23 163144

File: src/hooks/useGraphSettings.ts [hunk 6]
```diff
@@ -37,6 +42,22 @@ export const useGraphSettings = () => {
             weight: 'bold',
             size: 22
           }
+        },
+        grid: {
+          color: function (context: { tick: { value: number } }) {
+            if (context.tick.value !== 0) {
+              return GRAY_COLOR;
+            } else {
+              return fontColor;
```
Comment (JuliaShavyrova) [inline src/hooks/useGraphSettings.ts]: ternary operator would be more suitable here

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**